projects/frontend/cicd/.gitlab-ci.yml

```diff
@@ -148,6 +148,10 @@ frontend_publish_ui_image:
 
 frontend_deploy_testing_data_pipelines:
   stage: pre_release_test
```

Comment:
That is clearly from previous change but do you know why is "frontend_deploy_testing_data_pipelines" part fo pre_release_test stage? It should be part of pre_release stage where the actual "pre-releases" like testing deployment are done ?

Reply:
Yeah, makes sense. I must have missed it when I did the initial change.

Reply:
Looking at it again, here is the stage order.

```
  - build
  - publish_artifacts
  - pre_release
  - pre_release_image
  - pre_release_test
  - release
  - release_image
  - end
```

It's a bit strange, but it's set up so the image is published and tagged as `latest` in the `pre_release_image` stage. So it makes sense to deploy the image tagged with `latest` in the next stage. To be fair, it might be a better idea to move the initial publishing to `publish_artifacts` or `pre-release` and introduce an rc-tag. We can deploy and then tag with `latest` on subsequent stages, once we know the image is deployable, at least. This seems out of scope for this change, so I'll open a separate ticket.

Reply:
https://github.com/vmware/versatile-data-kit/issues/2377